source/routing/raptor_api.cpp

```diff
@@ -1184,6 +1184,22 @@ static std::vector<bt::ptime> parse_datetimes(RAPTOR& raptor,
     return datetimes;
 }
 
+static routing::map_stop_point_duration make_map_stop_point_duration(
+    const std::vector<type::EntryPoint>& entryPointList,
+    const std::unordered_map<std::string, type::StopPoint*>& raptor_stop_points_map) {
+    routing::map_stop_point_duration results;
+    for (const auto& entryPoint : entryPointList) {
+        auto it = raptor_stop_points_map.find(entryPoint.uri);
+        if (it != raptor_stop_points_map.end()) {
+            results[SpIdx{*it->second}] = navitia::seconds(entryPoint.access_duration);
+        } else {
+            // for now we throw, maybe we should ignore them
+            throw navitia::recoverable_exception("stop_point " + entryPoint.uri + " not found");
```

Comment:
You have a `map_find` utility to avoid a lot of boiler plate :
```cpp
    utils::make_map_find(raptor_stop_points_map, entryPoint.uri)
        .if_found([&](const type::StopPoint* sp){
            results[SpIdx{*sp}] = navitia::seconds(entryPoint.access_duration); 
        })
        .if_not_found([](){
            // for now we throw, maybe we should ignore them
            throw navitia::recoverable_exception("stop_point " + entryPoint.uri + " not found"); 
        });
```

Reply:
I'll try that, thanks!